src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1262,10 +1262,11 @@ internal virtual HttpResponseMessage GetResponse(HttpClient client, HttpRequestM
                     // For selected redirects that used POST, GET must be used with the
                     // redirected Location.
                     // Since GET is the default; POST only occurs when -Method POST is used.
-                    if (Method == WebRequestMethod.Post && IsRedirectToGet(response.StatusCode))
+                    if (req.Method == HttpMethod.Post && IsRedirectToGet(response.StatusCode))
```

Comment:
Please update comment too.
For the scenario `-CustomMethod "POST"` we should have redirection test(s) too.

Reply:
I see we have tests for POST redirection. So it is not clear where  abug is - in tests or in code.

Reply:
We never tested handling the redirect with -CustomMethod "POST", added tests

Reply:
I see now.

Reply:
> // Since GET is the default; POST only occurs when -Method POST is used.

Please update the comment.

Reply:
I already modified the comment in #18894, should I use that comment here ?

Reply:
Yes, we will merge the PR first. And I'd prefer more useful comment - code reader should understand that POST from parameter can be replaced on next step after redirection.

Reply:
done